tests/tensor/rewriting/test_basic.py

```diff
@@ -1237,33 +1237,98 @@ def test_local_join_1():
     assert len([n for n in e if isinstance(n.op, Join)]) == 0
     assert f.maker.fgraph.outputs[0].dtype == config.floatX
 
-    # test we don't apply when their is 2 inputs
+    # Test that join with 2 identical inputs gets optimized to tile
     s = join(1, a, a)
     f = function([a], s, mode=rewrite_mode)
     val = f([[1]])
-    assert np.all(val == [[1]])
+    assert np.all(val == [[1, 1]])  # joined along axis 1
     e = f.maker.fgraph.toposort()
-    assert len([n for n in e if isinstance(n.op, Join)]) == 1
+    assert len([n for n in e if isinstance(n.op, Join)]) == 0  # optimized away
```

Comment:
the spirit of this test condition was a case where the join remains. my suggestion was to do like we did to the other test and join(x x_reversed).

Otherwise this test is not about the old rewrite but the new rewrite and it's redundant with the new tests